src/mono/mono.proj

```diff
@@ -204,13 +203,17 @@
 
   <!-- Mono runtime build -->
   <Target Name="BuildMonoRuntime">
+    <PropertyGroup>
+      <_MonoLLVMTargetArchitecture Condition="'$(TargetArchitecture)' == 'wasm'">$(BuildArchitecture)</_MonoLLVMTargetArchitecture>
+      <_MonoLLVMTargetArchitecture Condition="'$(TargetArchitecture)' != 'wasm'">$(TargetArchitecture)</_MonoLLVMTargetArchitecture>
+    </PropertyGroup>
```

Comment:
I don't understand this magic.

I sort of expected monoaotcross to have the build/target logic and mono.proj would just look at RealTargetArchitecture every now and then

Reply:
mono.proj pretty much duplicates large sections - once for JIT, then again for AOT, with potentially differing flags passed to cmake and friends in both. `RealTargetArchitecture` (passed in by monoaotcross) is used in the AOT building section, see L743-744, but we don't have it in the JIT section.

And in the JIT, we need to special case wasm because it's the only case where building the JIT "for" architecture X (wasm) actually needs LLVM for architecture Y (build arch, i.e. x64). Basically we restored the LLVM need to build wasm into an `x64` folder, not into a `wasm` folder, so this variable needs to be overridden from `wasm` to `x64`

Reply:
@directhex  great explanation, thanks!  I wish we could capture this somewhere near the code.  either in a comment in the .proj, or in some (theoretical) docs/.../mono/cross-compiling.md design doc

Reply:
I keep starting a comment in the .proj, then end up in a multi-paragraph rant about how constrained we are by attempting to duplicate the 3-layer GNU build/target/host triad within msbuild. So I guess it needs to go in a design doc.